src/tests/Microsoft.PowerFx.Interpreter.Tests/TypeCoercionTest.cs

```diff
@@ -156,12 +169,31 @@ public void TryCoerceToRecordTest(string field1, string field2, bool expectedSuc
                 .Add(new NamedFormulaType(fieldName1, expectedFieldType1))
                 .Add(new NamedFormulaType(fieldName2, expectedFieldType2));
 
+            RecordType extraFieldTargetType = RecordType.Empty()
+                .Add(new NamedFormulaType(fieldName1, expectedFieldType1))
+                .Add(new NamedFormulaType(fieldName2, expectedFieldType2))
+                .Add(new NamedFormulaType(extraFieldName, FormulaType.String));
+
+            RecordType missingFieldTargetType = RecordType.Empty()
+                .Add(new NamedFormulaType(fieldName1, expectedFieldType1));
+
             bool isSucceeded = inputRecord.TryCoerceToRecord(targetType, out RecordValue result);
             if (expectedSucceeded)
             {
                 Assert.True(isSucceeded);
                 Assert.Equal(FormulaValue.New(double.Parse(field1)).Value, result.GetField(fieldName1).ToObject());
                 Assert.Equal(FormulaValue.New(bool.Parse(field2)).Value, result.GetField(fieldName2).ToObject());
+
+                bool isSucceededExtraField = inputRecord.TryCoerceToRecord(extraFieldTargetType, out RecordValue extraFieldResult);
+                Assert.True(isSucceededExtraField);
+                Assert.Equal(FormulaValue.New(double.Parse(field1)).Value, extraFieldResult.GetField(fieldName1).ToObject());
+                Assert.Equal(FormulaValue.New(bool.Parse(field2)).Value, result.GetField(fieldName2).ToObject());
+                Assert.Equal(FormulaType.Blank, extraFieldResult.GetField(extraFieldName).Type);
+
+                bool isSucceededMissingField = inputRecord.TryCoerceToRecord(missingFieldTargetType, out RecordValue missingFieldResult);
```

Comment:
>isSucceededMissingField [](http://example.com/codeflow?start=21&length=23)

Can we also check the Fields count? I expect it should be 1?

Reply:
Add count check.